src/Features/Lsif/Generator/Program.cs

```diff
@@ -92,14 +97,34 @@ private static async Task GenerateFromSolutionAsync(FileInfo solutionFile, ILsif
             }
 
             MSBuildLocator.RegisterInstance(msbuildInstance);
+        }
+
+        private static async Task GenerateFromProjectAsync(FileInfo projectFile, ILsifJsonWriter lsifWriter, TextWriter logFile)
+        {
+            await LocateAndRegisterMSBuild(logFile);
```

Comment:
Any reason not to just move the call to LocateAndRegisterMSBuild() into GenerateWithMSBuildLocatedAsync? Either merge the methods or just add it to the start of the other one, etc.

Reply:
Comments ont he method say: 

```
        // This method can't be loaded until we've registered MSBuild with MSBuildLocator, as otherwise
        // we load ILogger prematurely which breaks MSBuildLocator.
        [MethodImpl(MethodImplOptions.NoInlining)]
        private static async Task GenerateWithMSBuildLocatedAsync(
            FileInfo solutionOrProjectFile, ILsifJsonWriter lsifWriter, TextWriter logFile,
            Func<MSBuildWorkspace, Task<Solution>> openAsync)
        {
```